### PR TITLE
46-Fix-mypy-findings

### PR DESCRIPTION
**Pull request by rojberr**
#46

### CODE REVIEW

File: mypy.ini
Comment (rojberr) [inline mypy.ini:1]: There is no need to use `mypy.ini` cause we define mypy config in `tox.ini`;)

**Comment by rojberr** [comment]
I would start with:
- uncomment mypy code section in `.github/workflows/test-and-coverage.yml`
- try to change code and see how it impacts the results in the pipeline mypy step
- iterate until I solve all mypy findings ;)

**Comment by undeMalum** [comment]
I have added a section in `pyproject.toml` to ignore missing imports for libraries that do not support type hinting. This _reduced_ the number of errors. However, there are still a few things to consider:

- refactoring code: mypy seems to distinguish between e.g. `sklearn.metrics` and `sklearn` when it comes to throwing errors about missing imports. It poses a question: **Do we want to add also `sklearn.metric` to `pyproject.toml`, refactor the code to just use sklearn, or is there another way to cope with this?**
- new packages: some packages have their child version that support type hinting. Do we want to use them instead?

Tox run with mypy findings for reference: [here](https://github.com/GHOST-Science-Club/tree-classification-irim/actions/runs/14218238221/job/39839788236?pr=100#step:5:25).

**Comment by undeMalum** [comment]
All checks pass. _However_, my solution added OS-specific mypy environments. This is because if we supply tox with `-e` switch, then it overrides other tox environments and, in turn, mypy doesn't environment doesn't receive os environment and **packages do not get installed.**

File: tox.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tox]
 minversion = 4.0.0
-env_list = py{311,310}-{linux,windows}, flake8, mypy
+env_list = py{311,310}-{linux,windows}, flake8, mypy-{linux,windows}
```
Comment (undeMalum) [inline tox.ini:3]: All checks pass. _However_, my solution added OS-specific mypy environments. This is because if we supply tox with `-e` switch, then it overrides other tox environments and, in turn, mypy environment doesn't receive os environment and **packages do not get installed.**

**Review by undeMalum — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (rojberr) [reply tox.ini:3]: lgtm

File: .github/workflows/test-and-coverage.yml [hunk 2]
```diff
@@ -11,25 +11,36 @@ permissions:
...
+
+            -   name: Run tests with tox for Linux
+                if: runner.os == 'Linux'
+                run: tox -e mypy-linux
+    
+            -   name: Run tests with tox for Windows
+                if: runner.os == 'Windows'
+                run: tox -e mypy-windows
+
+            # -   name: Run TOX mypy
```
Comment (rojberr) [inline .github/workflows/test-and-coverage.yml:40]: Please remove this dead code :)

**Review by undeMalum — APPROVED** [review]
